File: Arjan_Niels/QL_Vizualizer/QL_Parser/AST/Nodes/QuestionNode.cs
namespace QLParser.AST.Nodes
{
    public class QuestionNode : Node
    {
        public string ID { get; private set; }
        public string Text { get; private set; }
        public QValueType ValueType { get; private set; }

        public QuestionNode(string id, string text, QValueType questionType) : base(NodeType.QUESTION)
        {
            this.ID = id;
            this.Text = text;
            this.ValueType = questionType;
        }

        public override string ToString()
        {
            return base.ToString() + string.Format("\tID: {0}\t{1}\t\t{2}", ID, ValueType, Text);
        }
    }
}

File: LDG_TV/SC-Java-LDG-TV/src/main/java/domain/model/QuestionNode.java
package domain.model;

import domain.model.variable.Variable;

public class QuestionNode extends Node {
    private String text;
    private Variable variable;

    public QuestionNode(String text, Variable variable) {
        this.text = text;
        this.variable = variable;
    }

    public String getText() {
        return text;
    }

    public Variable getVariable() {
        return variable;
    }



    @Override
    public String toString() {
        return this.text;
    }
}
